gabo 0.4.8 (new formula)

ashishb opened the pull request:
<!-- Use [x] to mark item done, or just click the checkboxes with device pointer -->

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

chenrui333 commented:
either squash and rebase to the latest master or create new branch from the latest master to have the file change.

chenrui333 commented:
also the file layout got changed, it needs to be `Formula/g/gabo.rb`

chenrui333 commented on Formula/gabo.rb:
```diff
@@ -0,0 +1,19 @@
+  depends_on "go" => :build
+
+  def install
+    system "make", "build_prod"
+    bin.install "src/gabo/bin/gabo" => "gabo"
```
```suggestion
    cd "src/gabo" do
      system "go", "build", *std_go_args(ldflags: "-s -w"), "./cmd/gabo"
    end
```

chenrui333 commented on Formula/gabo.rb:
```diff
@@ -0,0 +1,19 @@
+  end
+
+  test do
+    system "#{bin}/#{name}", "--help"
+    system "#{bin}/#{name}", "--version"
```
need better test than just version test

---

We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

ashishb replied:
I added tests to the codebase.
I am running them now instead.
See https://github.com/Homebrew/homebrew-core/pull/140306

ashishb commented:
> 
either squash and rebase to the latest master or create new branch from the latest master to have the file change.

Fixed

> 
also the file layout got changed, it needs to be Formula/g/gabo.rb

Fixed


> need better test than just version test

Pending

ashishb commented:
Superseded by https://github.com/Homebrew/homebrew-core/pull/140306
